New simple chilled water cooling panel

RKStrand opened the pull request:
Adds a new simple cooling panel, ZoneHVAC:CoolingPanel:RadiantConvective:Water.
The radiant/convective cooling panel is a component of zone equipment.  Chilled water is supplied by the plant and circulated through the panel.  Through radiation and convective heat transfer, the panel absorbs energy contained within the zone to provide cooling.  Control of the panel is achieved using the methodology used for low temperature radiant systems—a piecewise linear control profile that varies the flow rate to the panel based on a control parameter defined by the user.

@Myoldmopar @mjwitte @lgentile The simple chilled ceiling model that was completed last fall has now been reimplemented into the current develop and I believe that all issues should now be resolved.  There are some HVAC iteration maximums, but as this is a radiant system that is fairly lightweight, I think this is okay.  It isn't happening all of the time so this should be fine.
### Review Checklist

This will not be exhaustively relevant to every PR.
- [x] Code style (parentheses padding, variable names)
- [x] Functional code review (it has to work!)
- [x] CI status: all green or justified
- [x] Performance: CI Linux results include performance check -- verify this
- [x] Unit Test(s)
- C++ checks:
  - [x] Argument types
  - [x] If any virtual classes, ensure virtual destructor included, other things
- IDD changes:
  - [x] Verify naming conventions and styles, memos and notes and defaults
  - [x] Open windows IDF Editor with modified IDD to check for errors
- [x] If new idf included, locally check the err file and other outputs
- [x] Documentation changes in place
- [x] Changed docs build successfully

RKStrand committed:
Ported code changes to latest develop

All of the code for the implementation of the simple chilled ceiling
model have been brought forward to the current develop with this
commit.  Input files, IDD, and documentation changes to follow.

RKStrand committed:
IDF, IDD, Docs for Chilled Ceiling

The addition of the IDFs, the IDD, and the documentation for the new
simple chilled ceiling model.  Also includes a “bug” fix and the new
CMake to run the new IDFs.

RKStrand committed:
Final changes and corrections

Fixed a few bugs and other warning messages/issues, this is now ready
for final approval and merging

Myoldmopar committed:
Attempted merge back up to develop

Myoldmopar committed:
Code fixes to get it to build

Myoldmopar commented:
@RKStrand @mjwitte I merged develop in here, resolving many conflicts and fixing the required changes to get it to build.  At this point, the 9 test files for the cooling panel all run successfully.  I want to work on getting this merged in soon so that I never have to pull it back up again.

First question, can we trim down the number of test files so we don't add another 9 files to the CI runs, or do they need to be split up like they are?

Second question, nothing for now. I'll start reviewing...

RKStrand commented:
@myoldmopar

So glad to see that you are also thinking along the same lines...let's get this merged so we don't have to deal with it again.

In answer to your question, maybe the files could be cut back in number. It was done this way to test all of the options and make it easier to both create input files as well as compare differences in the various options. To reduce the number would mean we would have to manually go back and add chilled ceilings to other zones. You could maybe cut the number of example files down to two, but it would require changing the nodes, branches, loops, etc. to accommodate those changes. It made more sense during development to do it the way I did. Guess I wasn't worried about the number of test cases (more is better right?).

Myoldmopar commented on doc/src/docs/EngineeringReference/13e-EncyclopaedicRefs.md:
```diff
@@ -644,6 +644,115 @@ Several things should be noted about the fraction of heat that is radiated direc
+
+* In the event that condensation is predicted on the surface of a panel, the model uses the same condensation control options as the low temperature radiant system.  These include: ignoring the problem and allowing the simulation to continue, shutting the system down when condensation is predicted, and a variable response that tries to keep the system running at a lower output while avoiding cond ...
+
+* In order to avoid any violations of the Second Law of Thermodynamics or heating by the system, the cooling panel will be shut off if the inlet water temperature is above the zone temperature.
```
> In order to avoid any violations of the Second Law of Thermodynamics or heating by the system

I like this.  It's like you are saying:

In order to avoid **AN ALIEN INVASION**, or, you know, a spider

Myoldmopar commented on doc/src/docs/EngineeringReference/13e-EncyclopaedicRefs.md:
```diff
@@ -644,6 +644,115 @@ Several things should be noted about the fraction of heat that is radiated direc
+Calculating the UA value of a cooling panel or coming up with a relationship for such a term is not a simple mater.  However, most devices are either rated or tested by the manufacturer under certain conditions.  From this rating information, a UA value can be calculated.  The EnergyPlus model will request rating information (inlet water temperature, mass flow rate, “air” temperature, and panel o ...
+
+The development of the method for estimating the panel UA value will use the same equations shown above but simply with the purpose of calculating UA rather than the performance of the device, i.e. determining the output of the panel.  The method for estimating the UA value of the panel begins with the panel rating data that includes the following values:
+
```
mater -> matter

Myoldmopar commented on doc/src/docs/EngineeringReference/13e-EncyclopaedicRefs.md:
```diff
@@ -644,6 +644,115 @@ Several things should be noted about the fraction of heat that is radiated direc
+
+T_z=x_r MRT+(1-x_r )MAT 	(8)
+
+where xr is the “Fraction Radiant” for the radiant cooling panel that the user provides.
```
Probably need some blank lines between these paragraphs to get it to render properly.

Myoldmopar commented on doc/src/docs/InputOutputReference/01c-InputOutputReference.md:
```diff
@@ -15446,6 +15448,183 @@ This field reports the amount of radiant energy transferred from the baseboard t
+
+This field is the inlet temperature to the panel under conditions used to rate the capacity of the panel.  This value along with the other rated conditions is used to calculate the overall heat transfer coefficient (U-Value) that will be used to characterize the heat transfer between the water being circulated through the panel and the air in the space.  The default value is 5°C (44°F).
+
+#### Field: Rated Space Temperature
```
I wonder if `Zone` instead of `Space`?  Look at me doing idd field name review, @mjwitte :grimacing:?

RKStrand replied:
@Myoldmopar Sorry...thinking too much like an architecture person there and not an EnergyPlus developer.  I'll change that to ZONE...for this parameter and the next one as well which has the same issue.

Myoldmopar commented on doc/src/docs/InputOutputReference/01c-InputOutputReference.md:
```diff
@@ -15446,6 +15448,183 @@ This field reports the amount of radiant energy transferred from the baseboard t
* ZoneTotalLoad

Operative temperature for radiant system controls is the average of Mean Air Temperature and Mean Radiant Temperature. If the user does not select a control type, MeanAirTemperature control is assumed by EnergyPlus. See the throttling range and control temperature schedule fields below for more information.
+
```
Something looks super wonky with the bulleted list above.  Git thinks they are all on the same line??  Maybe a line ending problem?

RKStrand replied:
Yeah, not sure what the deal was there.  Must have been a weird line ending character or something.  Should be fixed now.

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -36464,6 +36464,776 @@ ZoneHVAC:Baseboard:RadiantConvective:Electric,
+ZoneHVAC:CoolingPanel:RadiantConvective:Water,
+  A1 , \field Name
+       \required-field
+  A2 , \field Available Schedule Name
```
~~Available~~ 

Availability

RKStrand replied:
@Myoldmopar Where appropriate with these changes to idd.in (this one and ones below), I am also updating the InputOutputReference section, if necessary.

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -36464,6 +36464,776 @@ ZoneHVAC:Baseboard:RadiantConvective:Electric,
+       \required-field
+       \type object-list
+       \object-list ScheduleNames
+  A3 , \field Inlet Node Name
```
~~Inlet Node Name~~

Water Inlet Node Name

Myoldmopar replied:
Oh, and these should have `\type node`

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -36464,6 +36464,776 @@ ZoneHVAC:Baseboard:RadiantConvective:Electric,
+       \object-list ScheduleNames
+  A3 , \field Inlet Node Name
+       \required-field
+  A4 , \field Outlet Node Name
```
~~Outlet Node Name~~

Water Outlet Node Name

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -36464,6 +36464,776 @@ ZoneHVAC:Baseboard:RadiantConvective:Electric,
+  N3 , \field Rated Water Mass Flow Rate
+       \type real
+       \minimum> 0.0
+       \units Kg/s
```
~~Kg/s~~

kg/s

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -36464,6 +36464,776 @@ ZoneHVAC:Baseboard:RadiantConvective:Electric,
+       \minimum> 0.0
+       \units Kg/s
+       \default 0.063
+  N4 , \field Rated Capacity 
```
Rated Cooling Capacity?

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -36464,6 +36464,776 @@ ZoneHVAC:Baseboard:RadiantConvective:Electric,
+       \type real
+       \units m3/s
+  A5 , \field Control Type
+       \note (Temperature on which unit is controlled)
```
Remove parentheses

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -36464,6 +36464,776 @@ ZoneHVAC:Baseboard:RadiantConvective:Electric,
+       \type real
+       \minimum 0
+       \maximum 1
+  N9 , \field Fraction of Radiant Energy Incident on People
```
Is there a decent default to use here?

RKStrand replied:
@Myoldmopar Well, not really because it will depend on the placement of the panel with respect to where people are at.  One would think that these would have a placement such that they would have a significant impact on the people in the space.  However, what exactly this number is will vary with each building.  So, putting a default in is really just picking a random number out of a hat...at least in my opinion.  Do you still want me to put a number in?

Myoldmopar commented on src/EnergyPlus/ChilledCeilingPanelSimple.cc:
```diff
@@ -0,0 +1,1460 @@
+	}
+
+	Real64
+	SumHATsurf( int const ZoneNum ) // Zone number
```
If this is an already implemented function elsewhere in E+, could we just pull that out into a public place and call it from multiple places so we don't have any duplicated code?

RKStrand replied:
@Myoldmopar Well, that is a good idea and would take a bit to implement because it is not just an issue for the new chilled ceiling but also every other system that has followed in the footsteps of the radiant model.  That's actually a lot of stuff including various baseboard models, ventilated slab, etc.  So, are you wanting this done for all of these now or just for the chilled ceiling or this handled as some sort of defect/improvement?  I definitely agree now that you have brought this up that it would be better from the standpoint of duplicated code to implement this as a publicly available function as I believe everything outside the surface heat balance routines follows the exact same method.

Myoldmopar commented on src/EnergyPlus/ChilledCeilingPanelSimple.hh:
```diff
@@ -0,0 +1,347 @@
+			ControlType( 0 ),
+			ColdSetptSchedPtr( 0 ),
+			CondCtrlType( 0 ),
+		    CondDewPtDeltaT( 0.0 ),
```
Indenting problem

RKStrand replied:
@Myoldmopar Weird...didn't see this in Xcode, but I deleted what was there and repositioned it again so that EquipType lines up with CondDewPtDeltaT.  Hopefully that resolves the problem?

Myoldmopar commented on src/EnergyPlus/ChilledCeilingPanelSimple.hh:
```diff
@@ -0,0 +1,347 @@
+			CoolingPanelInletTempFlowReSimIndex( 0 )
+		{}
+
+		// Member Constructor
```
We can actually remove the member constructor since we'll never actually call the constructor with all those values.

RKStrand replied:
@Myoldmopar So, basically remove everything from this line to the end of this particular definition?  This shows my lack of C++ knowledge because I have no idea what each of these different constructor sections does (still don't) and was simply following examples from modules I was familiar with.  Any ideas on where to find good descriptions of these?

RKStrand replied:
@Myoldmopar Ok, I deleted this.  Still seems to build.

Myoldmopar commented on src/EnergyPlus/ChilledCeilingPanelSimple.hh:
```diff
@@ -0,0 +1,347 @@
+	GetCoolingPanelInput();
+
+	void
+	InitCoolingPanel(
```
At least some of these functions should be packaged up as member functions of the above struct.  I can help you work that up.  That will also make it easier to work up some unit tests.

RKStrand replied:
@Myoldmopar Yes, I would appreciate help on working that up as well as understanding what that means.  I kinda of dislike feeling like I am an old dog trying to learn new tricks, but I DO want to figure this out so that I can continue to be a productive team member and don't mess a bunch of stuff up.

nrel-bot-2 commented:
@Myoldmopar @RKStrand @lgentile it has been 14 days since this pull request was last updated.

nrel-bot-3 commented:
@Myoldmopar @RKStrand @lgentile it has been 14 days since this pull request was last updated.

Myoldmopar commented:
I'm again tired of the CI messages; I'm going to close this, but we can re-open the PR at any time when it's ready to move forward again.

RKStrand commented:
@Myoldmopar The rest of the comments that I have not commented on are pretty straightforward and I plan to fix them as soon we've discussed the proper way to move forward.  So, hopefully none of this will take too long once I get things started.

RKStrand committed:
Merge branch 'develop' into 972154-NewSimpleChilledCeiling

RKStrand commented:
Reopening this pull request as I work towards getting all of the comments taken care of...

RKStrand committed:
Initial set of review changes

Numerous changes requested by Edwin’s review are addressed in this
commit as well as a change that was necessary due to changes in other
parts of the code since this work was rechecked in during October 2015

nrel-bot-3 commented:
@Myoldmopar @RKStrand @lgentile it has been 19 days since this pull request was last updated.

RKStrand committed:
Merge branch 'develop' into 972154-NewSimpleChilledCeiling

RKStrand committed:
Merge branch 'develop' into 972154-NewSimpleChilledCeiling

RKStrand committed:
Merge branch 'develop' into 972154-NewSimpleChilledCeiling

nrel-bot-3 commented:
@Myoldmopar @RKStrand @lgentile it has been 20 days since this pull request was last updated.

nrel-bot-3 commented:
@Myoldmopar @RKStrand @lgentile it has been 14 days since this pull request was last updated.

nrel-bot-3 commented:
@Myoldmopar @RKStrand @lgentile it has been 14 days since this pull request was last updated.

nrel-bot-3 commented:
@Myoldmopar @RKStrand @lgentile it has been 14 days since this pull request was last updated.

nrel-bot-3 commented:
@Myoldmopar @RKStrand @lgentile it has been 14 days since this pull request was last updated.

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 972154-NewSimpleChilledCeiling

Conflicts:
	doc/src/docs/EngineeringReference/13e-EncyclopaedicRefs.md
	doc/src/docs/InputOutputReference/01c-InputOutputReference.md

mjwitte commented on idd/Energy+.idd.in:
```diff
@@ -36472,6 +36472,778 @@ ZoneHVAC:Baseboard:RadiantConvective:Electric,
        \maximum 1
 
+ZoneHVAC:CoolingPanel:RadiantConvective:Water,
+  A1 , \field Name
```
@RKStrand Need these lines here:

```
       \extensible:2 - repeat last two fields, remembering to remove ; from "inner" fields.
       \memo The number of surfaces can be expanded beyond 100, if necessary, by adding more
       \memo groups to the end of the list
       \min-fields 18
```

mjwitte commented:
@RKStrand Working on merging this up and moving the docs into tex.  I took some liberties and added some text from similar units.

Comparing with other similar systems, there are a few changes that should be made:
- [ ] Replace the single "Rated Capacity" field with four fields to allow scalable sizing inputs, similar to ZoneHVAC:Baseboard:RadiantConvective:Water.
- [ ] Add the same checks/warnings for fraction radiant adding up to 1.0 that is being added in #5767

Wait until I've pushed up today's changes before doing any new work.

mjwitte committed:
Simple chilled ceiling - move docs from md to tex - part 1

mjwitte commented on src/EnergyPlus/ChilledCeilingPanelSimple.cc:
```diff
@@ -0,0 +1,1460 @@
+		using namespace DataHeatBalSurface;
+
+		// Return value
+		Rea
```
@RKStrand Need to replace this with the new copyright notice that now goes at the top of the file.  In hh file also.

mjwitte commented on testfiles/CMakeLists.txt:
```diff
@@ -32,6 +32,15 @@ ADD_SIMULATION_TEST(IDF_FILE 5ZoneAirCooled_UniformLoading.idf EPW_FILE USA_IL_C
+ADD_SIMULATION_TEST(IDF_FILE 5ZoneCoolingPanelBaseboardOWB.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
+ADD_SIMULATION_TEST(IDF_FILE 5ZoneCoolingPanelBaseboardSimpleOff.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
+ADD_SIMULATION_TEST(IDF_FILE 5ZoneCoolingPanelBaseboardTotalLoad.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
+ADD_SIMULATION_TEST(IDF_FILE 5ZoneCoolingPanelBaseboardVarOff.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
```
@RKStrand Nine new test files is too many.  Will need to consolidate or eliminate or fold into unit tests (of which there aren't any at the moment).

RKStrand replied:
@mjwitte Ok, I added a unit test and eliminated six of the input files.  The remaining input files are a standard run, one with zone load control, and one with variable condensation controls.  The new unit test verifies that the different temperature controls are getting the correct temperature which pretty much eliminates the need for the other input files.

mjwitte replied:
@RKStrand Thanks, that's much better.  Would it be feasible to consolidate into one file by using different controls in different zones?

RKStrand replied:
@mjwitte Well, it wouldn't be "easy" because each file has a cooling panel in only one of the systems.  I did that so that I could run each of the nine files and compare the results to each other to see what the cooling panel was doing in each case knowing that I was comparing the same zone across files.  Changing this would require me to go in and modify at least two more HVAC systems in the input file and I have to admit that could get a bit messy.  Can we say that three is a big improvement over nine and that the addition of a unit test was a nice side benefit? :)

mjwitte replied:
Sure, that's fine.

mjwitte commented:
@RKStrand @Myoldmopar This branch has been merged up, and a first pass made at moving doc from md to tex.  Didn't tackle any math or equations yet in engineering ref (feel free to . . . ).  Noticed some discrepancy between I/O Ref list of output variables and the source code.  I took the source code to be the final word and changed docs to match - needs a proofread.

RKStrand commented:
@mjwitte Thanks for getting this moving forward!  It will be good to get this into the code so that users have access to this model.  I will take a look through and address the comments that you made.  Hopefully this is now close to being done.

RKStrand commented:
@mjwitte One potential concern about the comment above regarding:
"Replace the single "Rated Capacity" field with four fields to allow scalable sizing inputs, similar to ZoneHVAC:Baseboard:RadiantConvective:Water."
Those inputs were based on a baseboard rating document that Daeho found that set the standard for baseboard manufacturers.  I did not use them here because I was not aware of any specific standard for cooling panels.  One manufacturer who has specs on-line actually refers to a DIN standard and the information they put out is closer to what is currently in the code rather than what is there for baseboards now.  The simple cooling panel did start with what was once the input for baseboards, but it seems to have changed.  The model is simple--it doesn't even ask for an area.  Do you still want this change, leave it as is, or come up with some middle ground?

RKStrand committed:
Change to IDD

Change to the IDD based on comments from @mjwitte

mjwitte commented:
@RKStrand I'm not talking about those fields, just the capacity method and related fields.  Look, instead at `ZoneHVAC:LowTemperatureRadiant:VariableFlow`:

```
  A10,  \field Cooling Design Capacity Method
        \type choice
        \key None
        \key CoolingDesignCapacity
        \key CapacityPerFloorArea
        \key FractionOfAutosizedCoolingCapacity
        \default CoolingDesignCapacity
        \note Enter the method used to determine the cooling design capacity for scalable sizing.
        \note CoolingDesignCapacity => selected when the design cooling capacity value is specified or
        \note auto-sized. CapacityPerFloorArea => selected when the design cooling capacity is determined
        \note from user specified cooling capacity per floor area and total floor area of cooled zone
        \note served by the hydrolic unit. FractionOfAutosizedCoolingCapacity => is selected when the
        \note design cooling capacity is determined from a user specified fraction and the auto-sized
        \note design cooling capacity of the system.
   N8 , \field Cooling Design Capacity
        \type real
        \units W
        \minimum 0.0
        \autosizable
        \note Enter the design cooling capacity. Required field when the cooling design capacity method
        \note CoolingDesignCapacity.
   N9 , \field Cooling Design Capacity Per Floor Area
        \type real
        \units W/m2
        \minimum 0.0
        \note Enter the cooling design capacity per total floor area of cooled zones served by the unit.
        \note Required field when the cooling design capacity method field is CapacityPerFloorArea.
   N10, \field Fraction of Autosized Cooling Design Capacity
        \type real
        \minimum 0.0
        \note Enter the fraction of auto-sized cooling design capacity. Required field when the cooling
        \note design capacity method field is FractionOfAutosizedCoolingCapacity.
```

Nearly all of the ZoneHVAC equipment has these scalable sizing field for heating/cooling capacity or for forced air zone equipment, they reference a `DesignSpecification:ZoneHVAC:Sizing` object.  It should be relatively easy to replicate the sizing code for this by copying from `ZoneHVAC:LowTemperatureRadiant:VariableFlow`.  @rraustad Can help with questions.

RKStrand committed:
Cooling panel radiant fraction change

Change made based on request by @mjwitte to make the radiant fraction
to surface summation checks similar to a recent defect fix in the high
temperature radiant heater so that no loss of energy is allowed.

RKStrand committed:
Addition of new copyright notices

Added the new copyright notices for 2016 to both the source code and
the header file for this work

RKStrand committed:
Creation of a unit test for new model

Created a unit test procedure for the new simple chilled ceiling model.
 Intent is also to reduce the number of input files in the test suite
as 9 new files was deemed too many.  This unit test will replace at
least 5 of the existing test files.

RKStrand committed:
Merge branch 'develop' into 972154-NewSimpleChilledCeiling

RKStrand committed:
Corrected issues with the unit test

Corrected a couple of problems with the unit tests.  With this commit,
the unit test is working properly and all tests are passing.

mjwitte commented on src/EnergyPlus/ChilledCeilingPanelSimple.cc:
```diff
@@ -0,0 +1,1571 @@
+
+		// REFERENCES:
+		// na
+
```
@RKStrand Please delete unused comment blocks like this.  Current style is minimal comments, don't need these template comments anymore.

mjwitte commented on src/EnergyPlus/ChilledCeilingPanelSimple.cc:
```diff
@@ -0,0 +1,1571 @@
+		//       AUTHOR         Rick Strand
+		//       DATE WRITTEN   February 2001
+		//       MODIFIED       Aug. 2007 Daeho Kang (Modification only for baseboard)
+		//  
```
Old copyright block here can be deleted.

mjwitte commented on src/EnergyPlus/ChilledCeilingPanelSimple.hh:
```diff
@@ -0,0 +1,302 @@
+	SumHATsurf( int const ZoneNum ); // Zone number
+
+	//*****************************************************************************************
+	//     NOTICE
```
And here - another old copyright block.

mjwitte commented on src/EnergyPlus/ZoneEquipmentManager.cc:
```diff
@@ -3263,6 +3265,12 @@ namespace ZoneEquipmentManager {
+					SimCoolingPanel( PrioritySimOrder( EquipTypeNum ).EquipName, ActualZoneNum, ControlledZoneNum, FirstHVACIteration, SysOutputProvided, ZoneEquipList( CurZoneEqNum ).EquipIndex( EquipPtr ) );
+					
+					NonAirSystemResponse( ActualZoneNum ) += SysOutputProvided;
+					LatOutputProvided = 0.0; // This baseboard does not add/remove any latent heat
```
Minor quibble - pasteo - "baseboard" --> "cooling panel"

mjwitte commented on src/EnergyPlus/ChilledCeilingPanelSimple.cc:
```diff
@@ -0,0 +1,1571 @@
+		//    INTEGER, PARAMETER   :: MaxDistribSurfaces    = 20         ! Maximum number of surfaces that a baseboard heater can radiate to
+		int const MinDistribSurfaces( 1 ); // Minimum number of surfaces that a baseboard heater can radiate to
+		Real64 const MinThrottlingRange( 0.5 ); // Smallest throttling range allowed in degrees Celsius
+		static std::string const MeanAirTemperature( "MeanAirTemperature" );
```
@RKStrand These don't need to be static.  In fact, they don't seem useful at all.  They are only used once, so why not put the literal strings in the ifs below?

mjwitte commented on src/EnergyPlus/ChilledCeilingPanelSimple.cc:
```diff
@@ -0,0 +1,1571 @@
+		// na
+
+		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+		static bool MyOneTimeFlag( true );
```
This should be move up to the top and renamed so it can be reset in a clear_state function.  Same for the getinput flag.

RKStrand committed:
Addition of autosizing options

Added some autosizing options as shown in other models and eliminated
some of the test input files that are no longer necessary due to the
addition of unit tests.  This builds but has not been vetted.

RKStrand committed:
Merge branch 'develop' into 972154-NewSimpleChilledCeiling

RKStrand committed:
Merge branch 'develop' into 972154-NewSimpleChilledCeiling

Myoldmopar commented:
@RKStrand From the conversation it looks like there are still a couple suggestions/notes from @mjwitte related to the static flags, etc.  While you are addressing those, note there are a couple build warnings.  They are related to unused variables, but they aren't quite that clear, since they are unused _using_ statements, and CI doesn't show exactly where they are defined but unused.  The two variables are 
- `EnergyPlus::DataHeatBalance::Zone`
- `EnergyPlus::DataSurfaces::TotSurfaces`

So as you make any further changes, the easiest thing to do is just search in your changes for mentions of those like:  `using DataHeatBalance::Zone` and you can either investigate which ones are used and which aren't, or better yet, comment them out and let the compiler tell you which ones were actually needed.

Alternatively, you could boot up a Linux VM and compile it to get a little more detail about the warning.

RKStrand committed:
Minor edits

Minor edits based on review comments by @mjwitte

RKStrand committed:
Merge branch 'develop' into 972154-NewSimpleChilledCeiling

RKStrand committed:
Changes for clear_state and unused uses

Changes to the code to implement a clear_state for the new model as
well as a comprehensive review of the using statements that included
removing unused variables and trying to push using statements down to
more local levels

RKStrand committed:
Merge branch 'develop' into 972154-NewSimpleChilledCeiling

mjwitte committed:
Chilled ceiling - Fix negative UA error

mjwitte committed:
Chilled ceiling - Fix line endings and version in new test files

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 972154-NewSimpleChilledCeiling
